### DISCUSSION TITLE
Send log to custom file / channel

### DISCUSSION THREAD

**Discussion by underdpt** [bilfeldt/laravel-http-client-logger#27] [Q&A] [answered]
Hello,

This may be pretty obvious, but I don't see how to send the logs to a custom file or log channel. Could you provide an example?

Thanks!

**Comment by bilfeldt** [comment, marked as answer]
@underdpt the packages default logger (can be [configured](https://github.com/bilfeldt/laravel-http-client-logger/blob/48013049c7398b933c2acc2c16c839cc3e9e3711/config/http-client-logger.php#L59)) can log to a filesystem by using these configurations:

https://github.com/bilfeldt/laravel-http-client-logger/blob/48013049c7398b933c2acc2c16c839cc3e9e3711/config/http-client-logger.php#L73C2-L86

**Reply by underdpt** [reply]
Thanks!

This is what I ended up with:

```php
$client = Http::baseUrl(config('ingram.endpoint'))
    ->logWhen(
        condition: app()->environment('dev'),
        config: ['channel' => 'ingram'],
    );
```

**Reply by bilfeldt** [reply]
That looks good 👍 

For the `condition` you could consider removing that part and simply set the environment variable `HTTP_CLIENT_LOGGER_FILTER_ALL=true` during development.
